src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -2846,58 +2847,110 @@ public void setSelectedNonDownloadableFiles(List<FileMetadata> selectedNonDownlo
+    }
+    
+    public String getZipDownloadLimitAsString(){
+        return FileSizeChecker.bytesToHumanReadable(settingsWrapper.getZipDownloadLimit());
+    }
+    
+    public Long getSizeOfSelectedOrigNumeric(){
+        boolean original = true;
+        return DatasetUtil.getDownloadSizeNumericBySelectedFiles(selectedFiles, original);
+        
+    }
+    
+    public Long getSizeOfSelectedArchivalNumeric(){
+        boolean original = false;
+        return DatasetUtil.getDownloadSizeNumericBySelectedFiles(selectedFiles, original);
+        
+    }
+    
+    public Long getSizeOfSelectedMaxNumeric(){
+       return Math.max(DatasetUtil.getDownloadSizeNumericBySelectedFiles(selectedFiles, true), DatasetUtil.getDownloadSizeNumericBySelectedFiles(selectedFiles, false)) ;       
+    }
+    
     public String getSizeOfDatasetOrig() {
         boolean original = true;
         return DatasetUtil.getDownloadSize(workingVersion, original);
     }
 
     public void validateAllFilesForDownloadArchival() {
+        this.setSelectedFiles(workingVersion.getFileMetadatas());
         boolean guestbookRequired = isDownloadPopupRequired();
         boolean downloadOriginal = false;
-        validateFilesForDownloadAll(guestbookRequired, downloadOriginal);
+        validateFilesForDownload(guestbookRequired, downloadOriginal);
     }
 
     /**
      * Can result in "requested optional service" error. For non-tabular files
      * it's safer to use validateAllFilesForDownloadArchival.
      */
     public void validateAllFilesForDownloadOriginal() {
+        //set all files as selected
+        this.setSelectedFiles(workingVersion.getFileMetadatas());
         boolean guestbookRequired = isDownloadPopupRequired();
         boolean downloadOriginal = true;
-        validateFilesForDownloadAll(guestbookRequired, downloadOriginal);
+        validateFilesForDownload(guestbookRequired, downloadOriginal);
     }
 
     public void validateFilesForDownload(boolean guestbookRequired, boolean downloadOriginal){
         setSelectedDownloadableFiles(new ArrayList<>());
         setSelectedNonDownloadableFiles(new ArrayList<>());
         
+        Long bytes = (long) 0;
+        
         if (this.selectedFiles.isEmpty()) {
-            //RequestContext requestContext = RequestContext.getCurrentInstance();
             PrimeFaces.current().executeScript("PF('selectFilesForDownload').show()");
             return;
         }
-        for (FileMetadata fmd : this.selectedFiles){
-            if(this.fileDownloadHelper.canDownloadFile(fmd)){
+        
+        for (FileMetadata fmd : this.selectedFiles) {
+            if (this.fileDownloadHelper.canDownloadFile(fmd)) {
                 getSelectedDownloadableFiles().add(fmd);
+                DataFile dataFile = fmd.getDataFile();
+                if (downloadOriginal && dataFile.isTabularData()) {
+                    bytes += dataFile.getOriginalFileSize() == null ? 0 : dataFile.getOriginalFileSize();
```

Comment:
OK, this is kind of weird. I copied the code from the Util without really thinking about it, but after thinking about it I thought it might be for cases of an old file where we didn't fill in or backfill the original file size - but then I looked at my local DB and I have a bunch of cases (well 5 out of 23) where the the original file size on the datatable record is null. I'm going to dig a little more.